$OpenBSD: patch-src_bcal_c,v 1.1.1.1 2018/11/08 04:23:43 bcallah Exp $

-q is a GNU bc flag to silence their very verbose startup message
Rename last to prev because bc(1) already has a variable named last

Index: src/bcal.c
--- src/bcal.c.orig
+++ src/bcal.c
@@ -204,7 +204,7 @@ static int try_bc(char *expr)
 		dup2(pipe_cp[1], STDOUT_FILENO); // Give stdout to parent
 		dup2(pipe_cp[1], STDERR_FILENO); // Give stderr to parent
 
-		ret = execlp("bc", "bc", "-q", (char*) NULL);
+		ret = execlp("bc", "bc", (char*) NULL);
 		log(ERROR, "execlp() failed!\n");
 		exit(ret);
 	}
@@ -213,7 +213,7 @@ static int try_bc(char *expr)
 	char buffer[128] = "";
 	ret = write(pipe_pc[1], "scale=10\n", 9);
 
-	ret = write(pipe_pc[1], "last=", 5);
+	ret = write(pipe_pc[1], "prev=", 5);
 	if (lastres.p[0])
 		ret = write(pipe_pc[1], lastres.p, strlen(lastres.p));
 	else
@@ -2084,7 +2084,7 @@ int main(int argc, char **argv)
 				case 'b':
 					cfg.bcmode ^= 1;
 					if (cfg.bcmode) {
-						printf("bc vars: scale = 10, ibase = 10, last = r\n");
+						printf("bc vars: scale = 10, ibase = 10, prev = r\n");
 						strncpy(prompt, "bc> ", 5);
 					} else
 						strncpy(prompt, "bcal> ", 7);
